include/pybind11/detail/common.h

```diff
@@ -635,177 +651,271 @@ using remove_cvref_t = typename remove_cvref<T>::type;
 using std::index_sequence;
 using std::make_index_sequence;
 #else
-template<size_t ...> struct index_sequence  { };
-template<size_t N, size_t ...S> struct make_index_sequence_impl : make_index_sequence_impl <N - 1, N - 1, S...> { };
-template<size_t ...S> struct make_index_sequence_impl <0, S...> { using type = index_sequence<S...>; };
-template<size_t N> using make_index_sequence = typename make_index_sequence_impl<N>::type;
+template <size_t...>
+struct index_sequence {};
```

Comment:
These sorts of things would be a lot more readable in the new form with an extra blank line in-between.